### ISSUE TITLE
Extremely high planning time for hypertables with lots of chunks when using LIMIT query (ChunkAppend)

### ISSUE THREAD

**Issue by andnig** [timescale/timescaledb#2897]
**Relevant system information:**
 - OS: CentOS 7.8
 - PostgreSQL version: 11.10
 - TimescaleDB version: 2.0.0
 - Installation method: using Docker

**Describe the bug**
Time-ordered Queries with Limit on a hypertable with 725 chunks (biggest chunks are about 3GB) are seemingly touching all chunks, even though more than enough rows (~5 Mio. per chunk) are available in the very first chunk to satisfy the limit query.

Hypertable dimension structure: (```SELECT * from timescaledb_information.dimensions```)
```
column_name |         column_type         | dimension_type | time_interval
----------------------------------------------------------------------------------
 timestamp   | timestamp with time zone    | Time           | 1 day         |
```
Indexes:
```
all btree
("timestamp", id, thing)
(thing, "timestamp" DESC)
("timestamp" DESC)
```

- Query: 
```SQL
select somecolumn, thing, timestamp from mytable order by timestamp desc limit 10;
```
- The query needs about 60 minutes to return (whereas planning requires the most portion of it)
- During the whole time, I see significant disk IO up to 900MB/s and than a drop to about 50MB/s
- Approx. half the chunks (the "older" ones) are located on a separate tablespace, physically located on a Blob Storage which is mounted to a cache file system (so each time one of the files gets accessed, it is loaded to the cache disk)
The "hot" chunks are on a storage with up to 900MB/s bandwidth
- Explain analyze output
(Please note: I removed hundreds of the never executed Index Scans in the below query plan. But as you might guess they were all similar). 
The Query plan shows, that the ChunkAppend itself works, however an extreme amount of planning-time is required. I guess, as the "old" chunks on the blob store need to be accessed.
```
                                                                                                              QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=0.43..0.79 rows=10 width=40) (actual time=110.322..110.531 rows=10 loops=1)
   Output: "mytable"."somecolumn", "mytable".thing, "mytable"."timestamp"
   Buffers: shared hit=220140 read=6
   ->  Custom Scan (ChunkAppend) on kuenz."mytable"  (cost=0.43..114655.83 rows=3146222 width=40) (actual time=110.320..110.527 rows=10 loops=1)
         Output: "mytable"."somecolumn", "mytable".thing, "mytable"."timestamp"
         Order: "mytable"."timestamp" DESC
         Startup Exclusion: false
         Runtime Exclusion: false
         Buffers: shared hit=220140 read=6
         ->  Index Scan using "_hyper_42_7886_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_7886_chunk  (cost=0.43..114655.83 rows=3146222 width=40) (actual time=110.318..110.422 rows=10 loops=1)
               Output: _hyper_42_7886_chunk."somecolumn", _hyper_42_7886_chunk.thing, _hyper_42_7886_chunk."timestamp"
               Buffers: shared hit=220140 read=6
         ->  Index Scan using "_hyper_42_7879_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_7879_chunk  (cost=0.43..266442.28 rows=5653934 width=40) (never executed)
               Output: _hyper_42_7879_chunk."somecolumn", _hyper_42_7879_chunk.thing, _hyper_42_7879_chunk."timestamp"
         ->  Index Scan using "_hyper_42_7874_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_7874_chunk  (cost=0.43..311754.80 rows=5453871 width=40) (never executed)
               Output: _hyper_42_7874_chunk."somecolumn", _hyper_42_7874_chunk.thing, _hyper_42_7874_chunk."timestamp"
         ->  Index Scan using "_hyper_42_7865_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_7865_chunk  (cost=0.43..129152.45 rows=3585389 width=39) (never executed)
               Output: _hyper_42_7865_chunk."somecolumn", _hyper_42_7865_chunk.thing, _hyper_42_7865_chunk."timestamp"
         ->  Index Scan using "_hyper_42_7861_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_7861_chunk  (cost=0.43..169120.56 rows=4663671 width=40) (never executed)
               Output: _hyper_42_7861_chunk."somecolumn", _hyper_42_7861_chunk.thing, _hyper_42_7861_chunk."timestamp"
         ->  Index Scan using "_hyper_42_7853_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_7853_chunk  (cost=0.43..336958.96 rows=6403626 width=40) (never executed)
               Output: _hyper_42_7853_chunk."somecolumn", _hyper_42_7853_chunk.thing, _hyper_42_7853_chunk."timestamp"
         ->  Index Scan using "_hyper_42_1004_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_1004_chunk  (cost=0.42..118068.73 rows=216925 width=36) (never executed)
               Output: _hyper_42_1004_chunk."somecolumn", _hyper_42_1004_chunk.thing, _hyper_42_1004_chunk."timestamp"
         ->  Index Scan using "_hyper_42_1003_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_1003_chunk  (cost=0.42..134295.36 rows=234618 width=36) (never executed)
               Output: _hyper_42_1003_chunk."somecolumn", _hyper_42_1003_chunk.thing, _hyper_42_1003_chunk."timestamp"
         ->  Index Scan using "_hyper_42_1000_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_1000_chunk  (cost=0.42..114734.18 rows=178989 width=36) (never executed)
               Output: _hyper_42_1000_chunk."somecolumn", _hyper_42_1000_chunk.thing, _hyper_42_1000_chunk."timestamp"
         ->  Index Scan using "_hyper_42_888_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_888_chunk  (cost=0.29..2863.29 rows=83260 width=36) (never executed)
               Output: _hyper_42_888_chunk."somecolumn", _hyper_42_888_chunk.thing, _hyper_42_888_chunk."timestamp"
         ->  Index Scan using "_hyper_42_889_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_889_chunk  (cost=0.29..2921.70 rows=84987 width=36) (never executed)
               Output: _hyper_42_889_chunk."somecolumn", _hyper_42_889_chunk.thing, _hyper_42_889_chunk."timestamp"
         ->  Index Scan using "_hyper_42_886_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_886_chunk  (cost=0.29..2715.60 rows=78827 width=37) (never executed)
               Output: _hyper_42_886_chunk."somecolumn", _hyper_42_886_chunk.thing, _hyper_42_886_chunk."timestamp"
         ->  Index Scan using "_hyper_42_887_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_887_chunk  (cost=0.42..5128.93 rows=148954 width=36) (never executed)
               Output: _hyper_42_887_chunk."somecolumn", _hyper_42_887_chunk.thing, _hyper_42_887_chunk."timestamp"
         ->  Index Scan using "_hyper_42_6557_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_6557_chunk  (cost=0.28..595.16 rows=983 width=39) (never executed)
               Output: _hyper_42_6557_chunk."somecolumn", _hyper_42_6557_chunk.thing, _hyper_42_6557_chunk."timestamp"
         ->  Index Scan using "_hyper_42_495_chunk_mytable_timestamp_idx" on _timescaledb_internal._hyper_42_495_chunk  (cost=0.28..560.47 rows=1370 width=34) (never executed)
               Output: _hyper_42_495_chunk."somecolumn", _hyper_42_495_chunk.thing, _hyper_42_495_chunk."timestamp"
 Planning Time: 3269528.495 ms
 Execution Time: 125.911 ms
(1462 rows)
```

For reference: If I ```select somecolumn, thing, timestamp from mytable where timestamp > '2021-02-03' order by timestamp desc;```, the query returns after a view seconds, as really only the recent chunk is touched (3378991 rows are returned)

**Expected behavior**
Only the most recent chunk is read, leaving all other chunks virtually untouched.

**Sidenote**
As the above problem and huge planning overhead originates and is most visible from our usage of the blob store tablespace, the same behavior can be seen, when using only the data on the hot storage (900MB/s SSD). The planning time is still about 50.000 - 100.000 times higher than the execution time.

**Comment by xddz9** [comment]
I ran into the same issue here.

Conceptually I think I understand it: TimescaleDB has no way of knowing which chunk a row belongs to if I don't specify a timestamp range, right? If I only needed the most recent row I suppose I could query `MAX(timestamp) WHERE id = ?` and then use that timestamp to narrow my query down to a single chunk, but if I need more than one row then we're back to scanning potentially thousands of chunks for just a handful of rows. (Note: I'm not particularly experienced with TimescaleDB or Postgres so my assumptions here may be completely off)

Is this expected behaviour that one is supposed to model around or is there anything else that could help speed up this kind of query?

**Comment by akuzm** [comment]
This should be fixed in 2.6. We made significant improvements to the planning time in that release.
